src/addrman.h

```diff
@@ -723,22 +729,19 @@ class CAddrMan
     CAddrInfo SelectTriedCollision_() EXCLUSIVE_LOCKS_REQUIRED(cs);
 
     //! Consistency check
-    void Check()
```

Comment:
Would it be a good idea to
1. do the check unconditionally at the end of `Unserialize()` (as if `m_consistency_check == 1`) and
2. if it fails, to treat it as disk corruption - throw `std::ios_base::failure` instead of `abort()`?

Reply:
Seems reasonable. This can be done as a follow-up if desired.